quickcomm/admin.py

```diff
@@ -1,13 +1,337 @@
+    model = Post
+    extra = 0
+    fields = ['title', 'description']
+    show_change_link = True
+    can_delete = False
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    @admin.display(description='Title')
+    def title(self, obj: Post):
+        return obj.title
+
+
+@admin.register(Author)
+class AuthorAdmin(admin.ModelAdmin):
+    list_display = ('display_name', 'host', 'location')
+    readonly_fields = ('host', 'external_url', 'location', 'profile_image_html')
+    actions_on_top = True
+    actions = ['sync_posts', 'clear_posts', 'sync_followers', 'clear_followers']
+    inlines = [PostInlineForAuthor]
+
+    change_form_template = 'admin/edit_author.html'
+
+    fieldsets = (
+        (None, {
+            'fields': ('display_name', 'github', 'profile_image', 'profile_image_html', 'location', 'user')
+        }),
+        ('Host Details', {
+            'fields': ('host', 'external_url')
+        }),
+    )
+
+    @admin.display(description='Display Name')
+    def display_name(self, obj: Author):
+        return obj.display_name
+
+    @admin.display(description="Profile Image")
+    def profile_image_html(self, obj: Author):
+        if obj is None or obj.profile_image is None:
+            return format(f"<p>No image found</p>")
+
+        return format_html(f"<img src={obj.profile_image} width=100pt height=100pt />")
+
+
+    @admin.display(description='Location of Author')
+    def location(self, obj: Author):
+        return obj.location
+
+
+    def has_add_permission(self, request, obj=None):
+        return True
+
+    # TODO fix permissions for HTML button actions
+    def has_change_permission(self, request, obj: Author=None):
+        if obj is None:
+            return True
+
+        if obj.is_local:
+            return True
+
+        if '_sync_posts' in request.POST or '_clear_posts' in request.POST:
+            return True
+
+        return False
+
+    @admin.action(description='Sync posts')
+    def sync_posts(self, request, queryset):
+        for author in queryset:
+            if author.is_local:
+                continue
+            sync_posts(author)
+
+
+    @admin.action(description='Clear posts')
+    def clear_posts(self, request, queryset):
+        for author in queryset:
+            Post.objects.filter(author=author).delete()
+
+    @admin.action(description='Sync Followers')
+    def sync_followers(self, request, queryset):
+        for author in queryset:
+            if author.is_local:
+                continue
+            sync_followers(author)
+
+    @admin.action(description='Clear Followers')
+    def clear_followers(self, request, queryset):
+        for author in queryset:
+            author.followers.all().delete()
+
+    # Taken from https://books.agiliq.com/projects/django-admin-cookbook/en/latest/custom_button.html
+    def response_change(self, request, obj: Host):
```

Comment:
do we need to update our README or something else to cite this?

Reply:
I'm not too sure, I think this is fine as is.